helm/charts/oathkeeper/templates/hpa-controller.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if and (eq .Values.global.ory.oathkeeper.maester.mode "controller") .Values.deployment.autoscaling.enabled }}
```

Comment:
I'm not sure this logic is required, we have a deployment which can be scaled in both cases. And I don't think we should bound hpa to an optional controller. Imho it would be better to have it like this in [hydra](https://github.com/ory/k8s/blob/master/helm/charts/hydra/templates/hpa.yaml)

Reply:
I added this because of the if statement in the `deployment-controller.yaml`. I can remove it if you want?

Reply:
The if in the `deployment-controller.yaml` controls if we would deploy the `maester` as a sidecar within the main oathkeeper deployment or not, but the end result in still one deployment for oathkeeper. We can safely attach the hpa to the deployment regardless of the maester setup ;)